src/guild.cpp

```diff
@@ -51,27 +55,23 @@ GuildRank_ptr Guild::getRankByLevel(uint8_t level) const
 	return nullptr;
 }
 
-void Guild::addRank(uint32_t rankId, std::string_view rankName, uint8_t level)
-{
-	ranks.emplace_back(std::make_shared<GuildRank>(rankId, rankName, level));
-}
-
-Guild* IOGuild::loadGuild(uint32_t guildId)
+Guild_ptr IOGuild::loadGuild(uint32_t guildId)
 {
 	Database& db = Database::getInstance();
-	if (DBResult_ptr result = db.storeQuery(fmt::format("SELECT `name` FROM `guilds` WHERE `id` = {:d}", guildId))) {
-		Guild* guild = new Guild(guildId, result->getString("name"));
+	DBResult_ptr result = db.storeQuery(fmt::format("SELECT `name` FROM `guilds` WHERE `id` = {:d}", guildId));
+	if (!result) {
+		return nullptr;
+	}
 
-		if ((result = db.storeQuery(
-		         fmt::format("SELECT `id`, `name`, `level` FROM `guild_ranks` WHERE `guild_id` = {:d}", guildId)))) {
-			do {
-				guild->addRank(result->getNumber<uint32_t>("id"), result->getString("name"),
-				               result->getNumber<uint16_t>("level"));
-			} while (result->next());
-		}
-		return guild;
+	const auto& guild = std::make_shared<Guild>(guildId, result->getString("name"));
+	if ((result = db.storeQuery(
+	         fmt::format("SELECT `id`, `name`, `level` FROM `guild_ranks` WHERE `guild_id` = {:d}", guildId)))) {
```

Comment:
No need to reuse the variable

```suggestion
	if (auto result = db.storeQuery(
	         fmt::format("SELECT `id`, `name`, `level` FROM `guild_ranks` WHERE `guild_id` = {:d}", guildId))) {
```

(likely needs reformatting)